### PR TITLE
feishin: fix darwin builds

### PR DESCRIPTION
**Pull request by jlbribeiro**
## Description of changes

* Fix Feishin's `darwin` builds, broken after #303638 (Electron "from source" packaging), as mentioned [here](https://github.com/NixOS/nixpkgs/pull/303638#issuecomment-2052463043). This takes an approach similar to #262081 and #280825.
* Disable update checks, as macOS builds didn't generate an `app-update.yml` after disabling code signing (and "direct" updates wouldn't be possible in any platform anyway).
* Added myself to the package's maintainers.

@lunik1, I kindly ask you to test this and see if it works on your `darwin` system now :) (including whether it plays any sound)

I tested it on a macOS Sonoma `x86_64-darwin` VM. The app only shows in Launchpad (but not on Applications nor Spotlight); I understand there are a couple of open issues in `nix-darwin` regarding _similar_ behavior, so I'm not 100% sure whether this is a packaging issue or just a Nix-on-macOS issue. I used `nix-darwin` with a very basic (i.e. no post-activation scripts + no Home Manager) config to install it. (Installed [`bruno`](https://search.nixos.org/packages?channel=unstable&type=packages&query=bruno) and it had the same issues.)

Maintainer: @onny

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lunik1** [comment]
It's working for me! Thanks!

**Comment by jlbribeiro** [comment]
@wegank would you be so kind to give this PR a look, similar to your https://github.com/NixOS/nixpkgs/pull/262081 which fixed `drawio` `darwin` builds?
(mine is also "inspired" by https://github.com/NixOS/nixpkgs/pull/280825)

File: pkgs/applications/audio/feishin/default.nix [hunk 1]
```diff
@@ -68,36 +77,58 @@ buildNpmPackage {
...
-        $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
-    done
+    ${
+      if stdenv.isDarwin then
+        ''
+          mkdir -p $out/{Applications,bin}
+          cp -r release/build/**/Feishin.app $out/Applications/
+          makeWrapper $out/Applications/Feishin.app/Contents/MacOS/Feishin $out/bin/feishin
+        ''
+      else
```
Comment (wegank) [inline pkgs/applications/audio/feishin/default.nix:108]:
```suggestion
  '' + lib.optionalString stdenv.isDarwin ''
    mkdir -p $out/{Applications,bin}
    cp -r release/build/**/Feishin.app $out/Applications/
    makeWrapper $out/Applications/Feishin.app/Contents/MacOS/Feishin $out/bin/feishin
  '' + lib.optionalString (!stdenv.isDarwin) ''
```

Please convert the `${if cond then a else b}` interpolations to string concatenations.

Comment (jlbribeiro) [reply pkgs/applications/audio/feishin/default.nix]: Thank you @wegank for your review; I did question which "version" to choose. Should be resolved now, hopefully.

**Comment by jlbribeiro** [comment]
@onny would you mind reviewing this PR? I've addressed @wegank's concerns (but rather not ping again given they're now RM for `24.05`), and would like to get the `darwin` build fixed. (Otherwise, I'd prefer to at least mark it as broken).

File: pkgs/applications/audio/feishin/default.nix [hunk 2]
```diff
@@ -123,8 +151,11 @@ buildNpmPackage {
     changelog = "https://github.com/jeffvli/feishin/releases/tag/v${version}";
     sourceProvenance = with sourceTypes; [ fromSource ];
     license = licenses.gpl3Plus;
-    platforms = platforms.unix;
+    inherit (electron.meta) platforms;
```
Comment (wegank) [inline pkgs/applications/audio/feishin/default.nix:154]:
```suggestion
    platforms = platforms.unix;
```

Looks like this caused the `8.has: clean-up` label. I guess `electron.meta` doesn't contain darwin when evaluated on `x86_64-linux`.

**Review by wegank — APPROVED** [review]

File: pkgs/applications/audio/feishin/default.nix (hunk 2 above)
Comment (jlbribeiro) [reply pkgs/applications/audio/feishin/default.nix]: @wegank Oh, interesting, thank you for catching that; I was trying to be consistent with other `electron` apps I'd seen.

Would you mind explaining why? Doesn't https://github.com/NixOS/nixpkgs/blob/7fd0e9b4456feebd79becd28aaa967343e49e74b/pkgs/development/tools/electron/binary/generic.nix#L41-L43 imply both `"x86_64-darwin"` and `"aarch64-darwin"` would be included?

I navigated the `platforms.*` code
https://github.com/NixOS/nixpkgs/blob/7fd0e9b4456feebd79becd28aaa967343e49e74b/lib/systems/doubles.nix#L110
https://github.com/NixOS/nixpkgs/blob/7fd0e9b4456feebd79becd28aaa967343e49e74b/lib/systems/inspect.nix#L93-L94
https://github.com/NixOS/nixpkgs/blob/3caeeb35f23c6f8ae1fcf2a57fbfa1fca19e5d58/lib/systems/parse.nix#L535
but didn't reach a satisfying conclusion.

I know you probably have too much on your plate, so if there's any resource on this _nuance_, please just link so :)
And again, thank you.

**Pull request merged**